clang/docs/checkedc/3C/INSTALL.md

```diff
@@ -0,0 +1,106 @@
+the following (from the root of the working tree) for a basic build:
+
+```
+(cd llvm/projects/checkedc-wrapper && git clone https://github.com/microsoft/checkedc)
+
+mkdir build && cd build
+cmake ../llvm -G Ninja -DLLVM_ENABLE_PROJECTS=clang
+ninja TARGET
+```
+
+where TARGET stands for the target(s) you want to build. For the `3c`
+command-line tool, the target is `3c`, but assuming you want to
+actually compile the Checked C source files generated by 3C, you'll
+also want to build `clang` (the Checked C compiler, which is a
+modified version of Clang). Executables will end up in `build/bin`;
+you'll likely want to add this directory to your `$PATH`.
+
+## Build options
+
+Because all of LLVM and clang are built as dependencies, this may
+consume up to ~50 GB of disk space, take several hours (depending on
+the speed of your computer), and require ~10 GB of RAM. Incremental
+builds will usually be faster. Here are some things you can do that
+may reduce the build time and/or peak memory use:
+
+- The above instructions already assume the use of the
+  [Ninja](https://ninja-build.org/) build tool; you may have to
+  install it. You can alternatively use `make` (remove `-G Ninja` from
+  the `cmake` command and replace `ninja` with `make`), but Ninja is
+  much faster in our experience.
+
+- Pass `-DLLVM_TARGETS_TO_BUILD=X86` (or whatever your machine's
+  architecture is, e.g., `ARM`) to `cmake` to build only the parts of
+  `clang` needed to compile programs for your machine.
+
+- Pass `-DLLVM_USE_LINKER=lld`. This requires a sufficiently recent
+  version of `lld` to be installed on your system.
+
+- Pass `-DLLVM_USE_SPLIT_DWARF=ON`. We have tested the build with this
+  option, but it may affect some debugging tools.
+
+- Pass `-DLLVM_OPTIMIZED_TABLEGEN=ON`.
+
+(The reference for improving build performance is [this LLVM
+page](https://www.llvm.org/docs/GettingStarted.html#common-problems),
+but we have attempted to describe the most promising options here.)
+
+You might want to use `-DCMAKE_BUILD_TYPE=RelWithDebInfo` if you are
+running 3C enough between builds that the faster running time
+outweighs the slower build time or you simply want to test a release
+build of 3C, assuming that's what end users will eventually be
+encouraged to use.
+
+## Mac OS X
+
+On some OS X versions, you have to add
+`-DDEFAULT_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk`
+to the `cmake` command; add it just after the `-DLLVM_ENABLE...`
+flags. You also need to make sure that the XCode developer
+command-line tools are installed, per [this
+thread](https://github.com/Homebrew/homebrew-core/issues/20791). Go to
+https://developer.apple.com/download/more/, log in with your Apple
+Developer ID, and download "Command Line Tools for Xcode 11.3.1".
+Then, enable these by invoking (from the command line), `sudo
+xcode-select -r`.
+
+## CR/LF issues
+
+There is a [known
+problem](https://github.com/correctcomputation/checkedc-clang/issues/317)
```

Comment:
We have a PR out to fix this: see #950. We ran into similar issues and tracked it down to .gitattributes as well.  This was introduced in an attempt to keep people from checking in files from Windows machines with accidental line feeds/carriage returns.  Git does not handle this well.

Reply:
Yes, I noticed the commit in connection with the LLVM upgrade but not the PR; thanks for the pointer.  I'm resolving this thread because it's not a problem with this PR.  I'll update our documentation as soon as feasible after you merge the fix.

Reply:
Oops, I see #950 was already merged.  Still, I want to get this PR merged ASAP and I hope it's OK if I wait to update our documentation until I have time to test your fix; that will be tracked by correctcomputation/checkedc-clang#317.

Reply:
By the way, I'm CCI's resident git expert (as you might have gathered) and would be happy to help you fix any line-ending-related problems in your workflows that remain after #950, subject to my other priorities.